src/main/webapp/tests/Components/FileInputComponent.test.jsx

```diff
@@ -111,11 +114,13 @@ test('Check that the proper error message is displayed for an invalid file type
     await userEvent.upload(fileInput, invalidFile);
 
     // Check that file is present
-    expect(screen.getByTestId('label')).toHaveTextContent('test.txt');
-    // Check if proper error msg is display
-    expect(screen.getByTestId('help')).toHaveTextContent('The file must be in json format');
-    // Check that handleChange is called with ##INVALID_FILE##
-    expect(handleChange).toHaveBeenCalledWith('testFileField', '##INVALID_FILE##');
+    await waitFor(() => {
+        expect(screen.getByTestId('label')).toHaveTextContent('test.txt');
+        // Check if proper error msg is display
+        expect(screen.getByTestId('help')).toHaveTextContent('The file must be in json format');
+        // Check that handleChange is called with ##INVALID_FILE##
+        expect(handleChange).toHaveBeenCalledWith('testFileField', '##INVALID_FILE##');
+    });
```

Comment:
minor one, but if you prefer to leave the waitFor statement i would use that same approach everywhere, so 
all excepts inside callback or just one, (here are 3 inside callback but in last wait for there is only one)